File: GameLibrary/src/cc/lib/zombicide/ui/UIZComponent.java
package cc.lib.zombicide.ui;

import cc.lib.game.AGraphics;
import cc.lib.ui.UIComponent;
import cc.lib.zombicide.ZTile;

public interface UIZComponent<T extends AGraphics> extends UIComponent {

    void loadTiles(T g, ZTile[] tiles);

}

File: Zombicide/src/cc/game/zombicide/ui/UIZComponent.kt
package cc.game.zombicide.ui

import cc.game.zombicide.ZQuest
import cc.game.zombicide.ZTile
import cc.lib.game.AGraphics
import cc.lib.ui.UIComponent

interface UIZComponent<in T : AGraphics> : UIComponent {
	fun loadTiles(g: T, tiles: Array<ZTile>, quest: ZQuest)
}